### PR TITLE
fix: mimic rejection for PuppeteerUtil on early call

### PR DESCRIPTION
**Pull request by jrandolf-google**
This PR rejects the PuppeteerUtil promise to mimic the behavior of when PuppeteerUtil cannot be evaluated due to navigation.

This also fixes a problem where the timeout would clear on navigation.

### CODE REVIEW

File: packages/puppeteer-core/src/common/WaitTask.ts [hunk 1]
```diff
@@ -170,11 +170,10 @@ export class WaitTask<T = unknown> {
   async terminate(error?: unknown): Promise<void> {
     this.#world.taskManager.delete(this);
 
-    if (this.#timeout) {
-      clearTimeout(this.#timeout);
-    }
-
     if (error && !this.#result.finished()) {
+      if (this.#timeout) {
```
Comment (OrKoN) [inline packages/puppeteer-core/src/common/WaitTask.ts:174]: is this change needed?

Comment (jrandolf) [reply packages/puppeteer-core/src/common/WaitTask.ts:174]: This could be a separate change, but I've made it here since it's small. Forgot to remark about it.

**Review by OrKoN — APPROVED** [review]
